### ISSUE TITLE
Users-Permissions password cannot be updated through regular API

### ISSUE THREAD

**Issue by thomas-br** [strapi/strapi#12275]
### Describe the bug

The password for a user entry of the users-permissions plugin cannot be updated through the regular API endpoint / plugins service. While the plugin is hashing the password in the update service, the field is marked as type `password`, thus gets encrypted _again_ in strapi's general entity-service (see [transforms.js](https://github.com/strapi/strapi/blob/master/packages/core/strapi/lib/services/entity-service/attributes/transforms.js)). When updated over the Content-Manager UI / API it works as expected, as not the strapi-core entity-service is used, but the one from the Content-Manager plugin instead.

### Steps to reproduce the behavior

1. Create a user in users-permissions over the Admin UI
2. Update the user and change it's password over the API `PUT /api/users/1`
3. Try to authenticate on the API `POST /api/auth/local` using the new password
4. See password failure

### Expected behavior

Updating user's password over API or the users-permissions services works and the updated password can be used for authentication

### Screenshots

`N/A`

### Code snippets

```JS
// in users-permission plugin
getService('user').edit(user.id, user); // will fail as password will be hashed twice

// first hash in users-permissions→services→user
if (params.password) {
      params.password = await getService('user').hashPassword(params);
    }

//users-permissions→services→user is using core entity-service to update
strapi.entityService.update

// entityService has a transform for fields of type password
```
### System

- Node.js version: 16
- NPM version: 6.14.15
- Strapi version: 4.0.4
- Database: sqlite
- Operating system: MacOS

### Additional context

Options I see:

- not use the core entity-manager for updating the users-permissions user entries
- [not using a field of type password](https://github.com/strapi/strapi/blob/7624cb2b3d195acf9465abf0a4645bfe1216bc4f/packages/plugins/users-permissions/server/content-types/user/index.js#L37), as the password is already hashed by the plugin. (Will this break setting a password over Content-Manager in Admin UI?)
- fully rely on the core entity-service's hashing and remove it from the plugin

**Comment by derrickmehaffy** [comment]
Related to the various bugs we have since U&P isn't using entityService

**Comment by Convly** [comment]
Fixed by https://github.com/strapi/strapi/pull/12276 (will be available with v4.0.7)
